### ISSUE TITLE
Connecting to remote ssh computer the next day always fails to connect the first login.

### ISSUE THREAD

**Issue by andyneff** [microsoft/vscode-remote-release#6265]
I've been dealing with this issue with SSH Remote-ing into my Linux box for over a year (if not years now). Every morning when I go to remote into my Linux machine from Windows, I get the following error:

![image](https://user-images.githubusercontent.com/7596961/152036152-4e19ebb5-cee5-42f2-921e-d8f41bca25a4.png)

But only once. It's always the very **first** time I try to connect. Every subsequent time I attempt to connect, it works. (I believe it works after the first successful time because the VS Code server executable is already running, and stay running until no connection for X minutes/hours, when it finally shuts itself down?)

```
$ ps -f `pgrep -f vscode`
UID          PID    PPID  C STIME TTY      STAT   TIME CMD
user      229132       1  0 16:48 ?        S      0:00 sh /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/server.sh --start-server --host=127.0.0.1 --enable-remote-auto-shutdown --port=0
user      229142  229132  0 16:48 ?        Sl     0:00 /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/node /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/out/vs/
user      229181  229142  0 16:48 ?        Sl     0:00 /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/node /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/out/boo
```

So everything works for the rest of the day, then at night, I get disconnected from VPN or laptop goes to sleep, and then in the morning again, it fails to connect the first time. I always have to log in twice first thing in the morning.

I've tried reloading VS Code on my local machine before trying to log in, but it always fails the first login

The only time it works the very first time, is when there's an update to VS Code, and I upgrade it before Remote SSH Connecting to my Linux machine. That gave me the idea that something in the `~/.vscode-server` dir is getting in a funky state.

- VSCode Version: 1.63.2, But it's been happening for well over a year now
- Local OS Version: Microsoft Windows 10.0.19044.1466
- Remote OS Version: Fedora 35 (but it's been happening for many fedora Versions now)
- Remote Filesystem for /home: btrfs
- Remote Extension/Connection Type: SSH
- Logs:

<details>
<summary>SSH Remote logs</summary>

```
[13:59:03.708] Log Level: 2
[13:59:03.708] remote-ssh@0.70.0
[13:59:03.708] win32 x64
[13:59:03.712] SSH Resolver called for "ssh-remote+remote", attempt 1
[13:59:03.712] "remote.SSH.useLocalServer": true
[13:59:03.712] "remote.SSH.path": C:\Users\user\.ssh\ssh.bat
[13:59:03.712] "remote.SSH.configFile": undefined
[13:59:03.712] "remote.SSH.useFlock": true
[13:59:03.713] "remote.SSH.lockfilesInTmp": false
[13:59:03.713] "remote.SSH.localServerDownload": auto
[13:59:03.713] "remote.SSH.remoteServerListenOnSocket": false
[13:59:03.713] "remote.SSH.showLoginTerminal": false
[13:59:03.713] "remote.SSH.defaultExtensions": []
[13:59:03.713] "remote.SSH.loglevel": 2
[13:59:03.713] "remote.SSH.serverPickPortsFromRange": {}
[13:59:03.713] "remote.SSH.enableDynamicForwarding": true
[13:59:03.713] "remote.SSH.serverInstallPath": {}
[13:59:03.714] SSH Resolver called for host: remote
[13:59:03.715] Setting up SSH remote "remote"
[13:59:03.718] Acquiring local install lock: C:\Users\user\AppData\Local\Temp\vscode-remote-ssh-83f61454-install.lock
[13:59:03.720] Looking for existing server data file at c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-ssh\vscode-ssh-host-83f61454-899d46d82c4c95423fb7e10e68eba52050e30ba3-0.70.0\data.json
[13:59:03.721] Using commit id "899d46d82c4c95423fb7e10e68eba52050e30ba3" and quality "stable" for server
[13:59:03.723] Install and start server if needed
[13:59:03.725] Checking ssh with "C:\Users\user\.ssh\ssh.bat -V"
[13:59:03.777] stdout> OpenSSH

[13:59:03.782] askpass server listening on \\.\pipe\vscode-ssh-askpass-a4db9643060d7cc96f305dc6cd42701eda8c539d-sock
[13:59:03.783] Spawning local server with {"serverId":1,"ipcHandlePath":"\\\\.\\pipe\\vscode-ssh-askpass-bc9a2899772a6ab47a906fa765f31d071a889d34-sock","sshCommand":"C:\\Users\\user\\.ssh\\ssh.bat","sshArgs":["-v","-T","-D","53609","remote","bash"],"dataFilePath":"c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-vscode-remote.remote-ssh\\vscode-ssh-host-83f61454-899d46d82c4c95423fb7e10e68eba52050e30ba3-0.70.0\\data.json"}
[13:59:03.783] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"c:\\Users\\user\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.70.0\\out\\local-server\\askpass.bat","VSCODE_SSH_ASKPASS_NODE":"C:\\Program Files\\Microsoft VS Code\\Code.exe","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"--ms-enable-electron-run-as-node","VSCODE_SSH_ASKPASS_MAIN":"c:\\Users\\user\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.70.0\\out\\askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"\\\\.\\pipe\\vscode-ssh-askpass-a4db9643060d7cc96f305dc6cd42701eda8c539d-sock"}
[13:59:03.790] Spawned 38796
[13:59:03.878] > local-server-1> Spawned ssh, pid=34560
[13:59:03.948] stderr> OpenSSH_8.8p1, OpenSSL 1.1.1l  24 Aug 2021
[13:59:04.041] stderr> debug1: Server host key: ssh-ed25519 SHA256:JR7eP7TUCLOoYh7sE6xJvV8rgKhbC4I1UwvhzARrKk0
[13:59:04.076] stderr> ********************************************************************************
[13:59:04.076] stderr>                                 NOTICE TO USERS
[13:59:04.076] stderr> 
[13:59:04.077] stderr> See the "Acceptable Use Policy" for further details
[13:59:04.077] stderr> 
[13:59:04.077] stderr> LOG OFF IMMEDIATELY if you do not agree to the conditions stated in this
[13:59:04.077] stderr> warning.
[13:59:04.077] stderr> 
[13:59:04.077] stderr> ********************************************************************************
[13:59:04.249] Got askpass request: {"request":"user@remote's password: "}
[13:59:04.249] Showing password prompt
[13:59:04.250] Listening for interwindow password on \\.\pipe\vscode-ssh-askpass-b57d4c230e2791f33625901b6ec21649ac379fa1-sock
[13:59:04.250] Writing password prompt to globalState
[13:59:09.655] Got password response
[13:59:09.655] Interactor gave response: ********************************************************
[13:59:09.655] Cleaning up other-window auth server
[13:59:10.525] stderr> Authenticated to remote ([*.*.*.*]:22) using "password".
[13:59:10.656] > ready: 515b06208c09
[13:59:10.663] > Linux 5.15.4-201.fc35.x86_64 #1 SMP Tue Nov 23 18:54:50 UTC 2021
[13:59:10.664] Platform: linux
[13:59:10.676] > 515b06208c09: running
[13:59:10.690] > Acquiring lock on /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/vscode-remote-lock.user.899d46d82c4c95423fb7e10e68eba52050e30ba3
[13:59:10.692] > Found existing installation at /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3...
[13:59:10.693] > Checking /home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log and /home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.pid for a running server
[13:59:10.694] > Looking for server with pid: 15513
[13:59:10.722] > Starting server with command... /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/server.sh --start-server --host=127.0.0.1 --enable-remote-auto-shutdown --port=0 --disable-telemetry   &> "/home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log" < /dev/null
> printenv:
[13:59:10.723] >     SHELL=/bin/bash
>     EDITOR=/usr/bin/nano
>     PWD=/home/user
>     KRB5CCNAME=KCM:
>     LOGNAME=user
>     XDG_SESSION_TYPE=tty
>     CUDA_INCLUDE_DIRS=/usr/include/cuda
>     MOTD_SHOWN=pam
>     HOME=/home/user
>     LANG=en_US.UTF-8
>     VSCODE_AGENT_FOLDER=/home/user/.vscode-server
>     CUDA_INC_PATH=/usr/include/cuda
>     SSH_CONNECTION=*.*.*.* 53610 *.*.*.* 22
>     MOZ_GMP_PATH=/usr/lib64/mozilla/plugins/gmp-gmpopenh264/system-installed
>     XDG_SESSION_CLASS=user
>     SELINUX_ROLE_REQUESTED=
>     LESSOPEN=||/usr/bin/lesspipe.sh %s
>     USER=user
>     SELINUX_USE_CURRENT_RANGE=
>     SHLVL=1
>     XDG_SESSION_ID=18
>     XDG_RUNTIME_DIR=/run/user/1033
>     SSH_CLIENT=*.*.*.* 53610 22
>     DEBUGINFOD_URLS=https://debuginfod.fedoraproject.org/ 
>     which_declare=declare -f
>     KDEDIRS=/usr
>     XDG_DATA_DIRS=/home/user/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share:/var/lib/snapd/desktop
>     PATH=/home/user/.local/bin:/home/user/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/var/lib/snapd/snap/bin
>     SELINUX_LEVEL_REQUESTED=
>     DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1033/bus
>     BASH_FUNC_which%%=() {  ( alias;
>      eval ${which_declare} ) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot "$@"
>     }
>     _=/usr/bin/printenv
[13:59:10.727] > Spawned remote server: 166761
[13:59:10.732] >  
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
[13:59:10.736] > Server did not start successfully. Full server log at /home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log >>>
[13:59:10.736] > <<< End of server log
> 515b06208c09: start
> exitCode==32==
> listeningOn====
> osReleaseId==fedora==
> arch==x86_64==
> tmpDir==/run/user/1033==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime==7==
> 515b06208c09: end
[13:59:10.736] Received install output: 
exitCode==32==
listeningOn====
osReleaseId==fedora==
arch==x86_64==
tmpDir==/run/user/1033==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime==7==

[13:59:10.737] Terminating local server
[13:59:10.739] Resolver error: Error: The VS Code Server failed to start
	at Function.ServerInstallError (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:430359)
	at u (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:424682)
	at Object.t.handleInstallOutput (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:429101)
	at Object.e [as tryInstallWithLocalServer] (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:469079)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:486771
	at async Object.t.withShowDetailsEvent (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:490561)
	at async c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:467603
	at async E (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:464111)
	at async Object.t.resolveWithLocalServer (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:467218)
	at async Object.t.resolve (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:488171)
	at async c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:564197
[13:59:10.741] ------




[13:59:10.743] Local server exit: null
```

</details>

# Steps to Reproduce:

## Method 1

Part of the reason it has taken to long to report this, is this has been very hard to reproduce realistically. Let's start with my actual use case, that involves waiting hours

1. Start VS Code
2. SSH Remote connect to a computer
3. Put the local computer to sleep
4. Come back the next day, and go to reload the windows and start a new connection.
5. The login will fail the first time:

![image](https://user-images.githubusercontent.com/7596961/152036152-4e19ebb5-cee5-42f2-921e-d8f41bca25a4.png)

## Method 2

Now, since I knew this problem occurs consistently every day, so I did the following:

1. Start VS Code
2. SSH Remote connect to a computer
3. Put the local computer to sleep
4. Come back the next day, do not reconnect VS Code
5. Ssh into the remote machine
6. `rsync -a ~/.vscode-server/ ~/.bad-vscode-server-state`

Now I'm able to reproduce the problem with ~50% success rate right away

1. Local machine: Make sure VS code and ssh.exe are not running.
2. Log into the remote machine
3. Remote: Kill all vscode server daemons `pkill -f vscode-server`
4. Remote: Restore the dir to a known broken state: `rsync -a ~/bad-vscode-server-state/ ~/.vscode-server` it has to be rsync (I'll try to explain below)
5. Local machine: Start VS Code, and login will fail the first time:

![image](https://user-images.githubusercontent.com/7596961/152036152-4e19ebb5-cee5-42f2-921e-d8f41bca25a4.png)

6. Trying a second time will succeed. Repeat starting at step 1 to recreate the error.

After a lot of digging, it _appears_ that removing `~/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log` will stop the first login from failing. This is where things start making less sense.

## Method 3

**Recreate the "bad log file":**

```
echo '

*
* Visual Studio Code Server
*
* Reminder: You may only use this software with Visual Studio family products,
* as described in the license https://aka.ms/vscode-remote/license
*

Extension host agent listening on 37013

[15:22:40] Extension host agent started.
[15:22:40] [127.0.0.1][6720a260][ManagementConnection] New connection established.
[15:22:40] [127.0.0.1][d30d8139][ExtensionHostConnection] New connection established.
[15:22:40] [127.0.0.1][d30d8139][ExtensionHostConnection] <15681> Launched Extension Host Process.' > ~/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log
``` 

However, this will not reproduce the error. Writing the file directly, or using `cp -a` or `scp -p` will not reproduce the error. Only when `rsync -a` is used, will there be a 50% chance of reproducing the problem.

1. Local: Close VS Code and ssh.exe (VS Code can leave a dangling ssh connection around, even when code.exe is no longer running)
2. Remote: Save the above log file to `~/.vscode-server/bad.log`, the sha256 sum should be `75d3e050ec9b2d698cc1355e293da9be4cba00e1840ed80fdaadf3d99c0c7ddb`
3. Remote: `pkill -f vscode-server`
4. Remote: `rsync -a ~/.vscode-server/bad.log ~/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log`, don't mess with the file after rsyncing it, even `touch`-ing will make it not fail.
5. Local: Restart VS Code and connect to the remote server and login will fail the first time:

![image](https://user-images.githubusercontent.com/7596961/152036152-4e19ebb5-cee5-42f2-921e-d8f41bca25a4.png)

6. Repeat starting at step 1 (Step 2 can be skipped)

---

Does this issue occur when you try this locally?: N/A?
Does this issue occur when you try this locally and all extensions are disabled?: N/A?

# Workarounds

1. Shutdown VS Code and VSCode-server
2. `rm ~/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log`
4. or `rm -r ~/.vscode-server`
5. Restart VSCode and connect

# Notes

The cause is **not** my `ssh.bat` script (which uses git for window's ssh executable. I can remove the following settings and it uses windows ssh client, and the exact same behavior occurs:

```
"remote.SSH.useLocalServer": true
"remote.SSH.path": C:\Users\user\.ssh\ssh.bat
```

---

I have had this issue working from several local machines (usually if not always windows) and several different remote machines. This is not specific to

**Comment by roblourens** [comment]
I'm not sure what the issue is, but your choice of filesystem might be suspicious given what you are saying here:

> Writing the file directly, or using cp -a or scp -p will not reproduce the error. Only when rsync -a is used, will there be a 50% chance of reproducing the problem.

Here is what happens when we connect:

1. Check for a running server
2. No server already running, so start the server with a command like `.../server.sh > .../.$version.log`
3. Wait for `.../.$version.log` to exist and contain "Extension host agent listening on..."
4. Read the file again to get the port number

This happens when you have a log file already present. So could this be happening?

1. Start the server, redirect its output to the log file
2. Start polling the log file
  1. At this point, the log file should be empty. But maybe your filesystem is slow to write or returns a cached copy when we read, so we get the contents of the old log file
3. We break out of the loop and read it again to get the port number
  1. At this point, the file is empty

The log file implies that series of events happened. The script did not wait at all, but was able to read the log file immediately after the `server.sh` command.

Does that make sense? Is your filesystem doing something wrong or am I wrong in how the redirection operator should work here?

**Comment by andyneff** [comment]
@roblourens I understand what you are saying, I don't know enough about btrfs to know it is possible or not. Normally I would say "That's not how I've observed it working". I looked at `/proc/{server.sh}/fd` and I do see exactly what you mentioned:

```
dr-x------. 2 user vsi  0 Feb  2 11:09 .
dr-xr-xr-x. 9 user vsi  0 Feb  2 11:09 ..
lr-x------. 1 user vsi 64 Feb  2 11:09 0 -> /dev/null
l-wx------. 1 user vsi 64 Feb  2 11:09 1 -> /home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log
l-wx------. 1 user vsi 64 Feb  2 11:09 2 -> /home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log
lr-x------. 1 user vsi 64 Feb  2 11:12 255 -> /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/server.sh
l-wx------. 1 user vsi 64 Feb  2 11:12 99 -> '/home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/vscode-remote-lock.user.899d46d82c4c95423fb7e10e68eba52050e30ba3 (deleted)'
```

My understanding of how bash/shell works is that the redirect will open the file with the file pointer at 0, before the bash process for the script "server.sh" even starts, as seen here:

**test.sh**

```bash
#!/usr/bin/env sh

cat blah.log >&2
date 2>&1
cat blah.log >&2
```

Running `./test.sh > blah.log` only shows the empty file, then the new time stamp, even if I do this before hand:

```
cp blah.log b2.log
rsync -a b2.log blah.log
./test.sh > blah.log
```

So it would seem that the files is "emptied" before the node js portion ever starts.

---

It's a standard btrfs install (for Fedora). That means `/` is a subvol, and `/home` is a subvol. Is there some info command you want me to run about how btrfs is setup?

I tried the following to see if it made a difference:

```bash
mv ~/.vscode-server ~/.vscode-server.bak
truncate -s 25G ~/vscode-server.img
sudo chown root:root ~/vscode-server.img
sudo chmod 600 ~/vscode-server.img
sudo mkfs.ext4 -j ~/vscode-server.img
mkdir ~/.vscode-server
sudo mount -t ext4 -o loop ~/vscode-server.img ~/.vscode-server
sudo chown user: ~/.vscode-server
rsync -a ~/.vscode-server.bak/ ~/.vscode-server
```

What this will do is create an ext3 "image file" that I mount onto `~/.vscode-server`, so for all practical purposes is it now "the other linux filesystem, ext3".

After doing this, I repeated "Method 3" above, and it did **error out again**. (only when using `rsync`, not `cp`). So ext3 did not seem to make a difference.

---

> We could easily delete the log file before this.

How can I test this out?

**Comment by andyneff** [comment]
I also gave this a try (for method 3)

```bash
rsync -a ~/.vscode-server/bad.log ~/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log
sudo sh -c "sync && echo 3 > /proc/sys/vm/drop_caches"
```

And the problem still occurs. That extra command "writes disk cache to disk", in case this was a weird OS caching issue, which is does not appear to be.

**Comment by andyneff** [comment]
I have verified that method 1 (the real issue), does in fact occur with the ext4 file system implemented [here](https://github.com/microsoft/vscode-remote-release/issues/6265#issuecomment-1028118598)

@tanhakabir What more information is needed?

**Comment by roblourens** [comment]
>>We could easily delete the log file before this.
>
>How can I test this out?

On our end, we could delete the log file before starting the server. Maybe I can make that change in our pre-release build tomorrow to test.

> I have verified that method 1 (the real issue), does in fact occur with the ext4 file system implemented https://github.com/microsoft/vscode-remote-release/issues/6265#issuecomment-1028118598

So you get that same error message? If you share the log, I can check if it's exactly the same series of events happening

**Comment by andyneff** [comment]
Method 1 while using EXT4 this morning:

<details><summary>SSH Logs</summary>

```
[08:33:35.397] Log Level: 2
[08:33:35.398] remote-ssh@0.70.0
[08:33:35.398] win32 x64
[08:33:35.401] SSH Resolver called for "ssh-remote+computer", attempt 1
[08:33:35.402] "remote.SSH.useLocalServer": true
[08:33:35.402] "remote.SSH.path": C:\Users\user\.ssh\ssh.bat
[08:33:35.402] "remote.SSH.configFile": undefined
[08:33:35.402] "remote.SSH.useFlock": true
[08:33:35.402] "remote.SSH.lockfilesInTmp": false
[08:33:35.402] "remote.SSH.localServerDownload": auto
[08:33:35.403] "remote.SSH.remoteServerListenOnSocket": false
[08:33:35.403] "remote.SSH.showLoginTerminal": false
[08:33:35.403] "remote.SSH.defaultExtensions": []
[08:33:35.403] "remote.SSH.loglevel": 2
[08:33:35.403] "remote.SSH.serverPickPortsFromRange": {}
[08:33:35.403] "remote.SSH.enableDynamicForwarding": true
[08:33:35.404] "remote.SSH.serverInstallPath": {}
[08:33:35.404] SSH Resolver called for host: computer
[08:33:35.404] Setting up SSH remote "computer"
[08:33:35.408] Acquiring local install lock: C:\Users\user\AppData\Local\Temp\vscode-remote-ssh-83f61454-install.lock
[08:33:35.410] Looking for existing server data file at c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-ssh\vscode-ssh-host-83f61454-899d46d82c4c95423fb7e10e68eba52050e30ba3-0.70.0\data.json
[08:33:35.411] Using commit id "899d46d82c4c95423fb7e10e68eba52050e30ba3" and quality "stable" for server
[08:33:35.414] Install and start server if needed
[08:33:35.416] Checking ssh with "C:\Users\user\.ssh\ssh.bat -V"
[08:33:35.480] stdout> OpenSSH

[08:33:35.485] askpass server listening on \\.\pipe\vscode-ssh-askpass-27043188a32965938145eac89a20ee6035277ca4-sock
[08:33:35.485] Spawning local server with {"serverId":1,"ipcHandlePath":"\\\\.\\pipe\\vscode-ssh-askpass-7aed902f56a002e9ce4cdf034a4d0a9da3a4e6b1-sock","sshCommand":"C:\\Users\\user\\.ssh\\ssh.bat","sshArgs":["-v","-T","-D","49625","computer","bash"],"dataFilePath":"c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-vscode-remote.remote-ssh\\vscode-ssh-host-83f61454-899d46d82c4c95423fb7e10e68eba52050e30ba3-0.70.0\\data.json"}
[08:33:35.486] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"c:\\Users\\user\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.70.0\\out\\local-server\\askpass.bat","VSCODE_SSH_ASKPASS_NODE":"C:\\Program Files\\Microsoft VS Code\\Code.exe","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"--ms-enable-electron-run-as-node","VSCODE_SSH_ASKPASS_MAIN":"c:\\Users\\user\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.70.0\\out\\askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"\\\\.\\pipe\\vscode-ssh-askpass-27043188a32965938145eac89a20ee6035277ca4-sock"}
[08:33:35.494] Spawned 52492
[08:33:35.598] > local-server-1> Spawned ssh, pid=64920
[08:33:35.746] stderr> OpenSSH_8.8p1, OpenSSL 1.1.1l  24 Aug 2021
[08:33:35.841] stderr> debug1: Server host key: ssh-ed25519 SHA256:JR7eP7TUCLOoYh7sE6xJvV8rgKhbC4I1UwvhzARrKk0
[08:33:35.878] stderr> ********************************************************************************
[08:33:35.878] stderr>                                 NOTICE TO USERS
[08:33:35.878] stderr> 
[08:33:35.878] stderr> 
[08:33:35.879] stderr> See the "Acceptable Use Policy" for further details
[08:33:35.879] stderr> 
[08:33:35.879] stderr> LOG OFF IMMEDIATELY if you do not agree to the conditions stated in this
[08:33:35.879] stderr> warning.
[08:33:35.879] stderr> 
[08:33:35.879] stderr> ********************************************************************************
[08:33:36.103] Got askpass request: {"request":"user@computer's password: "}
[08:33:36.104] Showing password prompt
[08:33:36.105] Listening for interwindow password on \\.\pipe\vscode-ssh-askpass-4490ff6bb6baeac88c1c404252cffd62f7ddc7b1-sock
[08:33:36.105] Writing password prompt to globalState
[08:33:44.039] Got password response
[08:33:44.039] Interactor gave response: ********************************************************
[08:33:44.039] Cleaning up other-window auth server
[08:33:44.915] stderr> Authenticated to computer ([*.*.*.*]:22) using "password".
[08:33:45.057] > ready: e7c8f20b4012
[08:33:45.066] > Linux 5.15.18-200.fc35.x86_64 #1 SMP Sat Jan 29 13:54:17 UTC 2022
[08:33:45.067] Platform: linux
[08:33:45.081] > e7c8f20b4012: running
[08:33:45.097] > Acquiring lock on /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/vscode-remote-lock.user.899d46d82c4c95423fb7e10e68eba52050e30ba3
[08:33:45.104] > Found existing installation at /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3...
> Checking /home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log and /home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.pid for a running server
[08:33:45.106] > Looking for server with pid: 1308635
[08:33:45.136] > Starting server with command... /home/user/.vscode-server/bin/899d46d82c4c95423fb7e10e68eba52050e30ba3/server.sh --start-server --host=127.0.0.1 --enable-remote-auto-shutdown --port=0 --disable-telemetry   &> "/home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log" < /dev/null
> printenv:
[08:33:45.139] >     SHELL=/bin/bash
>     EDITOR=/usr/bin/nano
>     PWD=/home/user
>     KRB5CCNAME=KCM:
>     LOGNAME=user
>     XDG_SESSION_TYPE=tty
>     CUDA_INCLUDE_DIRS=/usr/include/cuda
>     MOTD_SHOWN=pam
>     HOME=/home/user
>     LANG=en_US.UTF-8
>     VSCODE_AGENT_FOLDER=/home/user/.vscode-server
>     CUDA_INC_PATH=/usr/include/cuda
>     SSH_CONNECTION=*.*.*.* 49629 *.*.*.* 22
>     MOZ_GMP_PATH=/usr/lib64/mozilla/plugins/gmp-gmpopenh264/system-installed
>     XDG_SESSION_CLASS=user
>     SELINUX_ROLE_REQUESTED=
>     LESSOPEN=||/usr/bin/lesspipe.sh %s
>     USER=user
>     SELINUX_USE_CURRENT_RANGE=
>     SHLVL=1
>     XDG_SESSION_ID=19
>     XDG_RUNTIME_DIR=/run/user/1033
>     SSH_CLIENT=*.*.*.* 49629 22
>     DEBUGINFOD_URLS=https://debuginfod.fedoraproject.org/ 
>     which_declare=declare -f
>     KDEDIRS=/usr
>     XDG_DATA_DIRS=/home/user/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share:/var/lib/snapd/desktop
>     PATH=/home/user/.local/bin:/home/user/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/var/lib/snapd/snap/bin
>     SELINUX_LEVEL_REQUESTED=
>     DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1033/bus
>     BASH_FUNC_which%%=() {  ( alias;
>      eval ${which_declare} ) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot "$@"
>     }
>     _=/usr/bin/printenv
[08:33:45.147] > Spawned remote server: 1547022
[08:33:45.152] > *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
[08:33:45.157] > Server did not start successfully. Full server log at /home/user/.vscode-server/.899d46d82c4c95423fb7e10e68eba52050e30ba3.log >>>
[08:33:45.158] > <<< End of server log
> e7c8f20b4012: start
> exitCode==32==
> listeningOn====
[08:33:45.158] > osReleaseId==fedora==
> arch==x86_64==
> tmpDir==/run/user/1033==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime==12==
> e7c8f20b4012: end
[08:33:45.159] Received install output: 
exitCode==32==
listeningOn====osReleaseId==fedora==
arch==x86_64==
tmpDir==/run/user/1033==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime==12==

[08:33:45.159] Terminating local server
[08:33:45.161] Resolver error: Error: The VS Code Server failed to start
	at Function.ServerInstallError (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:430359)
	at u (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:424682)
	at Object.t.handleInstallOutput (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:429101)
	at Object.e [as tryInstallWithLocalServer] (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:469079)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:486771
	at async Object.t.withShowDetailsEvent (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:490561)
	at async c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:467603
	at async E (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:464111)
	at async Object.t.resolveWithLocalServer (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:467218)
	at async Object.t.resolve (c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:488171)
	at async c:\Users\user\.vscode\extensions\ms-vscode-remote.remote-ssh-0.70.0\out\extension.js:1:564197
[08:33:45.164] ------




[08:33:45.167] Local server exit: null
```

</details>

I was still running VSCode Version 1.63.2 when I got that error this morning

**Comment by roblourens** [comment]
That looks like the same thing, thanks for sharing.

I pushed a fix to delete the stale logfile before starting the server. We can see if that helps. You can "Switch to Pre-Release Version" which should be updated tomorrow morning with my change. If the code ran, you will see the message "Removing old logfile at"... in the log

**Comment by andyneff** [comment]
VSCode: 1.64.1
Remote SSH: v0.73.2022021015

```
[10:43:12.009] > Removing old logfile at /home/user/.vscode-server/.d6ee99e4c045a6716e5c653d7da8e9ae6f5a8b03.log
```

- [x] I am no longer able to reproduce the error with Method 3, on ext4 or btrfs
- [ ] I did not get the error this morning when I do my normal Method 1 login. I will report in a week if it never happens.

Thanks @roblourens!

**Comment by roblourens** [comment]
Sweet, let us know if you see this again

**Comment by andyneff** [comment]
I have gotten a "similar" error reconnecting to an ssh session the next day for the last two days. However, the log is different in that there is _no_ stack trace or error messages at all, and clicking "Reload windows" works without having to ssh log in again. So I'm calling that a "separate issue" and this issue fully resolved.
